sycl/source/detail/scheduler/commands.cpp

```diff
@@ -660,10 +660,12 @@ cl_int ExecCGCommand::enqueueImp() {
       case kernel_param_kind_t::kind_pointer:  {
         // TODO: Change to PI
         usesUSM = true;
+        std::shared_ptr<usm::USMDispatcher> USMDispatch =
+            getSyclObjImpl(Context)->getUSMDispatch();
         auto PtrToPtr = reinterpret_cast<intptr_t*>(Arg.MPtr);
         auto DerefPtr = reinterpret_cast<void*>(*PtrToPtr);
-        auto theKernel = pi::cast<cl_kernel>(Kernel);
-        CHECK_OCL_CODE(clSetKernelArgMemPointerINTEL(theKernel, Arg.MIndex, DerefPtr));
+        PI_CALL(
```

Comment:
You are not calling PI interface here, why using PI_CALL ?

Reply:
Because it was shorter than the cast to PI Result. :)